os/binfmt/binfmt_exit.c

```diff
@@ -129,13 +129,23 @@ int binfmt_exit(FAR struct binary_s *bin)
 		}
 		address = (FAR void *)sq_next((FAR sq_entry_t *)address);
 	}
-
+#ifdef CONFIG_OPTIMIZE_APP_RELOAD
+	if (!bin->reload)
+#endif
 	/* Free the RAM partition into which this app was loaded */
-	mm_free_ram_partition((uint32_t)bin->uheap);
+	mm_free_ram_partition((uint32_t)bin->ramstart);
 #endif
 	/* Free the load structure */
 
-	kmm_free(bin);
+#ifdef CONFIG_OPTIMIZE_APP_RELOAD
+	if (!bin->reload) {
+		kmm_free(bin->data_backup);
```

Comment:
`bin->data_backup` is just address which is in `roalloc` memory chunk. Is this really needed?

Reply:
Fixed